pkg/encryption/encryption_stores.go

```diff
@@ -0,0 +1,102 @@
+package encryption
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+// +kubebuilder:object:generate=true
+type Stores struct {
+	// ref to the secret that contains the keystore and its password
+	// the expected format of the secret is a "keystore" entry and a "keystore-password" entry
+	KeystoreSecretRef corev1.LocalObjectReference `json:"keystoreSecretRef"`
+
+	// ref to the secret that contains the truststore and its password
+	// the expected format of the secret is a "truststore" entry and a "truststore-password" entry
+	// +optional
+	TruststoreSecretRef *corev1.LocalObjectReference `json:"truststoreSecretRef"`
+}
+
+type EncryptionStoresYaml struct {
+	Keystore           string `json:"keystore,omitempty"`
+	KeystorePassword   string `json:"keystore_password,omitempty"`
+	Truststore         string `json:"truststore,omitempty"`
+	TruststorePassword string `json:"truststore_password,omitempty"`
+}
+
+type EncryptionStoresPasswords struct {
```

Comment:
I think I see a better option: we can get rid of this struct and store the passwords directly in `DatacenterConfig`:

```go
type DatacenterConfig struct {
...
	ServerEncryptionStores   *encryption.Stores
	ClientEncryptionStores   *encryption.Stores
	ClientKeystorePassword   string
	ClientTruststorePassword string
	ServerKeystorePassword   string
	ServerTruststorePassword string
}
```

Then you can simplify `ReadEncryptionStoresSecrets`:

```go
func ReadEncryptionStoresSecrets(ctx context.Context, klusterKey types.NamespacedName, template *DatacenterConfig, remoteClient client.Client, logger logr.Logger) error {
...
		if password, err := ReadEncryptionStorePassword(...); err != nil {
			return err
		} else {
			template.ClientKeystorePassword = password
		}
// etc
```

This will make it very easy to create the volume mounts.

For the JSON payload, you'd need to amend `CreateJsonConfig` but it seems easy.

Reply:
Note: as this involves a refactoring, we can do this in a follow-up task.